Projects/conf.sh

```diff
@@ -0,0 +1,174 @@
+END
+}
+
+ver_info() {
+	cat << END
+${SCRIPT} (TargetExpert builduils) ${VERSION}
+Copyright (c) 2019 Free Software Foundation.
+This is free software: you are free to change and redistribute it.
+There is NO WARRANTY, to the extent permitted by law.
+
+Written by Jinseong Jeon.
+END
+}
+
+parse_opts() {
+	local pf=0 cf=0 ff=0 tf=0
+
+	parse_lists=`getopt -o ${SOPTS} -l ${LOPTS} --name ${SCRIPT} -- "$@"`		#TODO: Error message duplicate
+	if [ $? -ne 0 -o $# -eq 0 ]; then						#	ex. "-ddd" option prints error message 3 times.
+		echo >&2 "Try '${SCRIPT} --help' for more information."
+		exit 255
+	fi
+	eval set -- ${parse_lists}
+
+	while true; do
+		case $1 in
+		-p|--project)
+			pf=1
+			case $2 in
+			=*|-*|''|' ') echo >&2 "${SCRIPT}: $2: invalid project name argument" && exit 255;;
+			esac
+			PROJ_NAME=$2 && shift 2
+			if [ ! -d ${CUR_PATH}/${PROJ_NAME} ]; then
+				echo >&2 "The project does not exist, Please create a project first."
+				exit 255;
+			fi ;;
+		-c|--core)
+			cf=1
+			case $2 in
+			=*|-*|''|' ') echo >&2 "${SCRIPT}: $2: invalid core name argument" && exit 255;;
+			esac
+			TG_MCU=$2 && shift 2 ;;
+		-f|--frequency)
+			ff=1
+			local chk_num=`echo $2 | sed s/[0-9]//g`
+			if [ ! -z $chk_num ]; then
+				echo >&2 "${SCRIPT}: $2: invalid frequency value argument"
+				exit 255
+			fi
+			case $2 in
+			''|' ') echo >&2 "${SCRIPT}: $2: invalid frequency value argument" && exit 255;;
+			esac
+			TG_FREQ=$2 && shift 2 ;;
+		-t|--type)
+			tf=1
+			case $2 in
+			=*|-*|''|' ') echo >&2 "${SCRIPT}: $2: invalid type name argument" && exit 255;;
+			esac
+			TG_TYPE=$2 && shift 2 ;;
+		-P|--port)
+			case $2 in
+			=*|-*|''|' ') echo >&2 "${SCRIPT}: $2: invalid port name argument" && exit 255;;
+			esac
+			TG_PORT=/dev/$2 && shift 2 ;;
+		-v|--version) ver_info && exit 0 ;;
+		-h|--help) usage && exit 0 ;;
+		--) shift ; break ;;
+		*) break ;;
+		esac
+	done
+
+	# Check the essential options.
+	if [ $((${pf}+${cf}+${ff}+${tf})) -ne 4 ]; then
+		echo >&2 "You must specify the '--project', '--core', '--frequency', '--type' options"
+		exit 255
+	fi
+}
+
+show_info() {
+	echo ""
+	echo "--------------------------------------------"
+	echo "  Your target configurations are as follow  "
+	echo "--------------------------------------------"
+	echo "* Project name : ${PROJ_NAME}"
+	echo "* MCU : ${TG_MCU}"
+	echo "* Frequency : ${TG_FREQ}HZ"
+	echo "* Type : ${TG_TYPE}"
+	echo "* Connected port : ${TG_PORT}"
+	echo "--------------------------------------------"
+	echo ""
+}
+
+create_cmakelist() {
+	local FILE_PATH=${PROJ_NAME}/CMakeLists.txt
+
+	cat << EOF > ${FILE_PATH}
+cmake_minimum_required(VERSION 3.14)
+	
+include(TESuit/TEInclude.cmake)
+
+Set_Target(${TG_MCU})
+Set_TargetFreq(${TG_FREQ})
+Set_JointInterface(${TG_TYPE})
+Set_JointPort(${TG_PORT})
+Set_Project(${PROJ_NAME})
+
+Init_TE()
+Prebuild_TE()
+
+project(\${Project_Name} \${Language})
+add_executable(\${Project_Name} main.c \${Src_Files})
```

Comment:
nit:
I have a little idea for improving this script. In many cases developer makes a directory like '. c', '.h' file. 
How about automatically putting the .c/.h file in this line?

Reply:
Thanks, your comment.
It`s a good idea, i`ll consider how to add it to the script.